Core/Object Arts/Dolphin/IDE/Base/InternalIcon.cls

```diff
@@ -182,12 +182,55 @@ imageManager
 	^IconImageManager current!
 
 new
-	^self withExtent: self defaultExtent! !
+	^self withExtent: self defaultExtent!
+
+stbConvert: anArray fromVersion: anInteger 
+	"Private - Convert from earlier version view by updating and answering the array of instance
+	variables, instVarArray. "
+
+	| instVars |
+	instVars := anArray.
+	anInteger < 1 ifTrue: [instVars := self stbConvertFromVersion0: instVars].
+	^instVars!
+
+stbConvertFrom: anSTBClassFormat 
```

Comment:
I have a question about the object upgrade process. If I have a subclass of Model, can I have my own version numbers and stbConvertFrom: methods? It seems that the answer would be "no", but that makes subclassing from Model somewhat less useful. Right?